pkgs/kde/third-party/krohnkite/default.nix

```
@@ -0,0 +1,56 @@
+{
+  lib,
+  buildNpmPackage,
+  fetchFromGitHub,
+  nodejs_22,
+  kwin,
+  kpackage,
+  zip,
+}:
+buildNpmPackage rec {
+  pname = "krohnkite";
+  version = "0.9.7";
+
+  src = fetchFromGitHub {
+    owner = "anametologin";
+    repo = "krohnkite";
+    rev = "refs/tags/${version}";
+    hash = "sha256-8A3zW5tK8jK9fSxYx28b8uXGsvxEoUYybU0GaMD2LNw=";
+  };
+
+  npmDepsHash = "sha256-My1goFEoZW9kFA3zb8xKPxAPXm6bypyq+ajPM8zVOHQ=";
+
+  dontWrapQtApps = true;
+
+  nodejs = nodejs_22;
+
+  nativeBuildInputs = [
+    kpackage
+    zip
+    kwin
+  ];
+
+  postPatch = ''
+    cp ${./package-lock.json} package-lock.json
+  '';
+
+  npmBuildScript = "tsc";
+
+  installPhase = ''
+    runHook preInstall
+
+    substituteInPlace Makefile --replace-fail '7z a -tzip' 'zip -r'
+    make krohnkite-${version}.kwinscript
+    kpackagetool6 --type=KWin/Script --install=krohnkite-${version}.kwinscript --packageroot=$out/share/kwin/scripts
```

relying on the attr `version` inside `installPhase` is making it cumbersome to override `src`, doing a typical version override like the following fails with `make: *** No rule to make target 'kronhkite-0.9.7.kwinscript'.`

```
(kdePackages.krohnkite.overrideAttrs (_oldAttrs: let
        rev = "0.9.8.2";
      in {
        version = "v${rev}";
        src = fetchFromGitHub {
          inherit rev;
          owner = "anametologin";
          repo = "krohnkite";
          hash = "sha256-chADfJ1zaufnwi4jHbEN1Oec3XFNw0YsZxLFhnY3T9w=";
        };
      }))
```

So I tried to override the `installPhase` to remove the dependency on `version` as follows and was able to successfully build 0.9.8.2, @Ben9986 wdyt of this change? I can create a PR to add it if you don't mind.
```suggestion
    make KWINPKG_FILE=krohnkite.kwinscript krohnkite.kwinscript
    kpackagetool6 --type=KWin/Script --install=krohnkite.kwinscript --packageroot=$out/share/kwin/scripts
```

---

Thanks for noticing this. I'll update the package and have a proper look at this over the next couple days.

---

#348145